theano/tensor/basic.py

```diff
@@ -5419,3 +5420,87 @@ def perform(self, node, inputs, (z, )):
         choice = inputs[1]
         # TODO reuse out?
         z[0] = numpy.choose(a, choice, mode=self.mode)
+
+
+class AllocEmpty(gof.Op):
+    """Implement Alloc on the gpu, but without initializing memory."""
+    __props__ = ()
+
+    # specify the type of the data
+    def __init__(self, dtype):
+        assert isinstance(dtype, string)
+        self.dtype = 'NPY_' + dtype.upper()
+
+    @staticmethod
+    def validate_shape(shape):
+        sh = [tensor.as_tensor_variable(s) for s in shape]
+        bcast = []
+        for s in sh:
+            if s.type.dtype[:3] not in ('int', 'uin'):
+                raise TypeError('Shape arguments must be integers', s)
+            # if s is constant 1, then we're broadcastable in that dim
+            try:
+                const_shp = tensor.get_scalar_constant_value(s)
+            except tensor.NotScalarConstantError:
+                const_shp = None
+            bcast.append(numpy.all(1 == const_shp))
+        otype = tensor.TensorType(dtype='float32', broadcastable=bcast)
+        output = otype()
+        return sh, output
+
+    def make_node(self, *shape):
+        shape, output = self.validate_shape(shape)
+        output.tag.values_eq_approx = tensor.type.values_eq_approx_always_true
+        return Apply(self, shape, [output])
+
+    def perform(self, node, inputs, out_):
+        out, = out_
+        sh = tuple([int(i) for i in inputs])
+        if out[0] is None or out[0].shape != sh:
+            # XXX: We could implement and call CudaNdarray.empty(sh) instead.
+            out[0] = numpy.zeros(sh)
```

Comment:
This line initializes the output elements to zeros instead of allocating an empty array. You can use numpy.empty to allocate a memory region without initialization